web-api/src/personalization/routes.rs

```diff
@@ -495,29 +505,29 @@ struct SemanticSearchResponse {
 
 async fn semantic_search(
     state: Data<AppState>,
-    document_id: Path<String>,
-    query: Query<SemanticSearchQuery>,
+    query: Json<UnvalidatedSemanticSearchQuery>,
```

Comment:
:art: we could destructure it already here to avoid the `.into_inner()` call below, like we do in other routes
```suggestion
    Json(query): Json<UnvalidatedSemanticSearchQuery>,
```